### PR TITLE
Add specialization for double for interp function

### PR DESCRIPTION
**Pull request by jeanlaroche**
I specialized interp for numeric types. Let me know if the way I did it isn't righ.

### CODE REVIEW

File: pythran/pythonic/numpy/interp.hpp [hunk 1]
```diff
@@ -1,20 +1,20 @@
-#ifndef PYTHONIC_NUMPY_INTERP_HPP
-#define PYTHONIC_NUMPY_INTERP_HPP
+#ifndef PYTHONIC_NUMPY_INTERPHPP
```
Comment (serge-sans-paille) [inline pythran/pythonic/numpy/interp.hpp:1]: This change is surprising

File: pythran/pythonic/include/numpy/interp.hpp [hunk 2]
```diff
@@ -13,38 +13,76 @@ namespace numpy
 
   // None,None,None
   template <class T1, class T2, class T3>
-  types::ndarray<double, types::pshape<long>>
+  typename std::enable_if<!std::is_arithmetic<T1>::value,
```
Comment (serge-sans-paille) [inline pythran/pythonic/include/numpy/interp.hpp:16]: do you mean std::is_integral here?

Comment (serge-sans-paille) [reply pythran/pythonic/include/numpy/interp.hpp:16]: ok, after discussing it on IRC, this change looks good, but doesn't capture `std::complex`. Is that intended?

Comment (jeanlaroche) [reply pythran/pythonic/include/numpy/interp.hpp:16]: mmm no, I just didn't think of it. I should probably add a specialization for complex.

**Comment by serge-sans-paille** [comment]
Just renamed out_type into interp_out_type, and removed the template alias redeclaration in the implementation file.

**Comment by serge-sans-paille** [comment]
Great job, thanks a lot @jeanlaroche

**Pull request merged**